### ISSUE TITLE
Guideline for Default value

### ISSUE THREAD

**Issue by fnobilia** [RADAR-CNS/RADAR-Schemas#45]
We should define a guideline about `default` values.

In AVRO, `default` values are crucial to ensure schema evolution. Quoting the [Avro Specification 1.8.2](https://avro.apache.org/docs/current/spec.html)
> if the reader's record schema has a field that contains a default value, and writer's schema does not have a field with the same name, then the reader should use the default value from its field. If the reader's record schema has a field with no default value, and writer's schema does not have a field with the same name, an error is signalled.

In case of UNION with `null`, AVRO provides two approaches:
- `optional` values, the default value is `null` because the UNION is defined like `"type": ["null", "string"]`
- `nullable` value, the default value class must match the no-null schema. For instance in case `"type": ["string", "null"]` the default value must be a `string`
For union the default value must be an instance of the first type appearing in the UNION definition.

In case of ENUM, we can force a symbol `UNKNOWN` as default value.

In all other cases, the policy to apply is left to the developer.

Applying the correct policy can also simplify the downstream analysis.

**Comment by fnobilia** [comment]
@mbegalevibrent waiting for your update

**Comment by blootsvoets** [comment]
Another reason not to prefer `NaN` as a default double value: the `AvroData.fromConnectSchema` chokes on it:

```
org.apache.avro.SchemaBuilderException: org.codehaus.jackson.JsonParseException: Non-standard token 'NaN': enable JsonParser.Feature.ALLOW_NON_NUMERIC_NUMBERS to allow
  at [Source: java.io.StringReader@94a86f7; line: 1, column: 7]
  at org.apache.avro.SchemaBuilder.toJsonNode(SchemaBuilder.java:2595)
  at org.apache.avro.SchemaBuilder.access$4900(SchemaBuilder.java:217)
  at org.apache.avro.SchemaBuilder$FieldBuilder.completeField(SchemaBuilder.java:2115)
  at org.apache.avro.SchemaBuilder$FieldBuilder.access$5300(SchemaBuilder.java:2034)
  at org.apache.avro.SchemaBuilder$GenericDefault.withDefault(SchemaBuilder.java:2423)
  at io.confluent.connect.avro.AvroData.addAvroRecordField(AvroData.java:679)
  at io.confluent.connect.avro.AvroData.fromConnectSchema(AvroData.java:613)
```

**Comment by blootsvoets** [comment]
Closed with #55
